cloudflare/resource_cloudflare_record_test.go

```diff
@@ -323,6 +323,49 @@ func TestAccCloudflareRecord_CreateAfterManualDestroy(t *testing.T) {
 	})
 }
 
+func TestAccCloudflareRecord_TtlValidation(t *testing.T) {
+	t.Parallel()
+	domain := os.Getenv("CLOUDFLARE_DOMAIN")
+	recordName := "tf-acctest-ttl-validation"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareRecordDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCheckCloudflareRecordConfigTtlValidation(domain, recordName),
+				ExpectError: regexp.MustCompile(fmt.Sprintf("error validating record %s: ttl cannot be set when `proxied` is true", recordName)),
+			},
+		},
+	})
+}
+
+func TestAccCloudflareRecord_TtlValidationUpdate(t *testing.T) {
+	t.Parallel()
+	domain := os.Getenv("CLOUDFLARE_DOMAIN")
+	recordName := "tf-acctest-ttl-validation"
+	name := "cloudflare_record.foobar"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareRecordDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareRecordConfigProxied(domain, recordName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareRecordExists(name, &cloudflare.DNSRecord{}),
+				),
+			},
+			{
+				Config:      testAccCheckCloudflareRecordConfigTtlValidation(domain, recordName),
+				ExpectError: regexp.MustCompile(fmt.Sprintf("error validating record %s: ttl cannot be set when `proxied` is true", recordName)),
```

Comment:
Since we've updated the error message, we'll also need to update this assertion.

```suggestion
				ExpectError: regexp.MustCompile(fmt.Sprintf("error validating record %s: ttl must be set to 1 when `proxied` is true", recordName)),
```